src/components/Files/PreUploadModal.tsx

```diff
@@ -98,25 +110,20 @@ export const PreUploadDialog = ({
         (e.target as HTMLInputElement).files as FileList,
       );
 
-      let incorrectSizeFiles: string[] = [];
-      let incorrectTypeFiles: string[] = [];
-      const filteredFiles = files.reduce<File[]>((acc, file) => {
-        const fileType = file.name.slice(file.name.lastIndexOf('.'));
-        if (file.size > 512 * bytesInMb) {
-          incorrectSizeFiles = incorrectSizeFiles.concat(file.name);
-          return acc;
-        }
-        if (
-          !allowedTypes.includes('*/*') &&
-          allowedTypes.length &&
-          !allowedTypes.includes(fileType)
-        ) {
-          incorrectTypeFiles = incorrectTypeFiles.concat(file.name);
-          return acc;
-        }
+      const incorrectSizeFiles: string[] = getFilesWithInvalidFileSize(
+        files,
+        512 * bytesInMb,
+      ).map((file) => file.name);
+      const incorrectTypeFiles: string[] = getFilesWithInvalidFileType(
+        files,
+        allowedTypes,
+      ).map((file) => file.name);
+      const filteredFiles = files.filter(
+        (file) =>
+          !incorrectSizeFiles.includes(file.name) &&
+          !incorrectTypeFiles.includes(file.name),
```

Comment:
wrap `incorrectSizeFiles` and `incorrectTypeFiles` into `new Set` to search with O(1)
```suggestion
          !incorrectSizeFiles.has(file.name) &&
          !incorrectTypeFiles.has(file.name),
```